src/Fixer/ClassNotation/StaticPrivateMethodFixer.php

```diff
@@ -0,0 +1,298 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class StaticPrivateMethodFixer extends AbstractFixer
+{
+    /**
+     * @var array
+     */
+    private $magicNames = [
+        '__clone' => true,
+        '__construct' => true,
+        '__destruct' => true,
+        '__serialize' => true,
+        '__set_state' => true,
+        '__sleep' => true,
+        '__unserialize' => true,
+        '__wakeup' => true,
+    ];
```

Comment:
What about other magic methods like `__call`? They can't be `static` either, right?

Reply:
I started with all magic methods in the test cases, but some like `__call` cannot be private (https://3v4l.org/gNl6u) and therefore cannot be added into them.

If needed I can still add all of them into the array, but cannot be tested.